src/Confluent.Kafka/Impl/LibRdKafka.cs

```diff
@@ -80,106 +113,215 @@ static void LoadNet45Librdkafka()
                 throw new InvalidOperationException(additionalMessage, win32Exception);
             }
         }
-#endif
+#else
 
-        static LibRdKafka()
+        static IntPtr TryLoadLibrdkafka(string librdkafkaName, string librarySuffix, LibraryOpenDelegate openLib)
         {
-#if NET45 || NET46
-            try
+            string confluentKafkaAssemblyLocation = typeof(Error).GetTypeInfo().Assembly.Location;
+            if (!confluentKafkaAssemblyLocation.EndsWith("Confluent.Kafka.dll"))
```

Comment:
Compare with OrdinalIgnoreCase. From experience, some scripts of deployment tends to use lower case everytime.